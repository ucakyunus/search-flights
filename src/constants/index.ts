export enum Currency {
  TRY = 'TRY',
}

export enum FareCategories {
  BUSINESS = 'BUSINESS',
  ECONOMY = 'ECONOMY',
}

export enum BrandCode {
  ECO_FLY = 'ecoFly',
  EXTRA_FLY = 'extraFly',
  PRIME_FLY = 'primeFly',
}

export enum Status {
  AVAILABLE = 'AVAILABLE',
  ERROR = 'ERROR',
}

export enum ListSortTypes {
  PriceAscending = 'PriceAscending',
  PriceDescending = 'PriceDescending',
  DepartureTimeAscending = 'DepartureTimeAscending',
  DepartureTimeDescending = 'DepartureTimeDescending',
}

export enum ListSortBy {
  DepartureTime = 'DepartureTime',
  Price = 'Price',
}

export enum ResultTypes {
  Success = 'Success',
  Failure = 'Failure',
}

export enum CounterActions {
  Increment = 'INCREMENT',
  Decrement = 'DECREMENT',
}
