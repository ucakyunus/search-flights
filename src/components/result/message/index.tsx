import { Box, Typography } from '@mui/material';
import { ResultTypes } from '@/constants';
import { CheckCircle, Cancel } from '@mui/icons-material';

export interface MessageProps {
  state: ResultTypes;
}

export function Message({ state }: MessageProps) {
  const icon =
    state === ResultTypes.Success ? (
      <CheckCircle color='success' sx={{ fontSize: 30 }} />
    ) : (
      <Cancel color='error' sx={{ fontSize: 30 }} />
    );

  const messageText =
    state === ResultTypes.Success
      ? 'Kabin seciminiz tamamlandi.'
      : 'Kabin seciminiz tamamlanamadi.';

  return (
    <Box py={2} display='flex' borderBottom={1} alignItems='center' borderColor='secondary.light'>
      {icon}
      <Typography ml={1} fontSize={16} fontWeight={500} color='primary.dark'>
        {messageText}
      </Typography>
    </Box>
  );
}
