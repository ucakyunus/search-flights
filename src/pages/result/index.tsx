import React, { useEffect } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { ResultTypes } from '@/constants';
import withSuspense from '@/components/lazy';
import { Box } from '@mui/material';

const Failure = withSuspense(React.lazy(() => import('@/components/result/failure')));
const Success = withSuspense(React.lazy(() => import('@/components/result/success')));

const Result = () => {
  const location = useLocation();
  const navigate = useNavigate();

  useEffect(() => {
    if (!location.state) {
      navigate('/');
    }
  }, [location, navigate]);

  if (!location.state || typeof location.state === 'undefined') {
    return null;
  }

  const { status, amount, currency } = location.state;

  return (
    <Box mt={6}>
      {status === ResultTypes.Success ? (
        <Success amount={amount} currency={currency} />
      ) : (
        <Failure />
      )}
    </Box>
  );
};

export default Result;
