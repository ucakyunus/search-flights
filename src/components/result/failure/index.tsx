import { Box, Button } from '@mui/material';
import { useNavigate } from 'react-router-dom';

import { Message } from '@/components/result/message';
import { ResultTypes } from '@/constants';

const Failure = () => {
  const navigate = useNavigate();

  const onClickHandler = () => {
    navigate('/');
  };

  return (
    <Box display='flex' flexDirection='column'>
      <Message state={ResultTypes.Failure} />

      <Button
        onClick={onClickHandler}
        variant='contained'
        color='error'
        sx={{ py: 1, px: 4, fontSize: 12, mt: 2, ml: 'auto' }}
      >
        Başa Dön
      </Button>
    </Box>
  );
};

export default Failure;
