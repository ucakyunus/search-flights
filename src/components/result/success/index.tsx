import { Box, Typography } from '@mui/material';

import { Message } from '@/components/result/message';
import { ResultTypes } from '@/constants';

export interface SuccessProps {
  amount: number;
  currency: string;
}

const Success = ({ amount, currency }: SuccessProps) => {
  const price = `${currency} ${amount} `;

  return (
    <Box>
      <Message state={ResultTypes.Success} />
      <Box display='flex' alignItems='center' justifyContent='space-between' pt={2}>
        <Typography fontSize={26} fontWeight={300} color='secondary.main'>
          Toplam Tutar
        </Typography>
        <Typography fontSize={26} color='darkcyan'>
          {price}
        </Typography>
      </Box>
    </Box>
  );
};

export default Success;
