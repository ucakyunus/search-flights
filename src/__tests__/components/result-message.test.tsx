import { render } from '@testing-library/react';
import { Message, MessageProps } from '@/components/result/message';
import { ResultTypes } from '@/constants';

const mockPropsSuccess: MessageProps = {
  state: ResultTypes.Success,
};

const mockPropsFailure: MessageProps = {
  state: ResultTypes.Failure,
};

test('Message component matches snapshot for success state', () => {
  const { asFragment } = render(<Message {...mockPropsSuccess} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Message component matches snapshot for failure state', () => {
  const { asFragment } = render(<Message {...mockPropsFailure} />);
  expect(asFragment()).toMatchSnapshot();
});
