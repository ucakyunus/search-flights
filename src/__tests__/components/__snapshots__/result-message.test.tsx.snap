// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component matches snapshot for failure state 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1ri905o"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1a4ayes-MuiSvgIcon-root"
      data-testid="CancelIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2m5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 css-zp3nss-MuiTypography-root"
    >
      Kabin seciminiz tamamlanamadi.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Message component matches snapshot for success state 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1ri905o"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-y0m7b1-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 css-zp3nss-MuiTypography-root"
    >
      Kabin seciminiz tamamlandi.
    </p>
  </div>
</DocumentFragment>
`;
